task/sast-snyk-check-oci-ta/0.3/sast-snyk-check-oci-ta.yaml

```diff
@@ -0,0 +1,262 @@
+        - name: PROJECT_NAME
+          value: $(params.PROJECT_NAME)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+        - name: COMPONENT_LABEL
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.labels['appstudio.openshift.io/component']
+      script: |
+        #!/usr/bin/env bash
+
+        set -euo pipefail
+        # shellcheck source=/dev/null
+        . /utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+
+        if [[ -z "${PROJECT_NAME}" ]]; then
+            PROJECT_NAME=${COMPONENT_LABEL}
+        fi
+
+        echo "The PROJECT_NAME used is: ${PROJECT_NAME}"
+
+        # Installation of Red Hat certificates for cloning Red Hat internal repositories
+        ca_bundle=/mnt/trusted-ca/ca-bundle.crt
+        if [ -f "$ca_bundle" ]; then
+          echo "INFO: Using mounted CA bundle: $ca_bundle"
+          cp -vf $ca_bundle /etc/pki/ca-trust/source/anchors
+          update-ca-trust
+        fi
+
+        SNYK_TOKEN_PATH="/etc/secrets/snyk_token"
+        if [ -f "${SNYK_TOKEN_PATH}" ] && [ -s "${SNYK_TOKEN_PATH}" ]; then
+          # SNYK token is provided
+          SNYK_TOKEN="$(cat ${SNYK_TOKEN_PATH})"
+          export SNYK_TOKEN
+        else
+          # According to shellcheck documentation, the following error can be ignored as it is ignored through indirection: https://www.shellcheck.net/wiki/SC2034
+          # shellcheck disable=SC2034
+          to_enable_snyk='[here](https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/testing_applications/enable_snyk_check_for_a_product/)'
+          note="Task $(context.task.name) skipped: If you wish to use the Snyk code SAST task, please create a secret name snyk-secret with the key 'snyk_token' containing the Snyk token by following the steps given ${to_enable_snyk}"
+          TEST_OUTPUT=$(make_result_json -r SKIPPED -t "$note")
+          echo "${TEST_OUTPUT}" | tee "$(results.TEST_OUTPUT.path)"
+          exit 0
+        fi
+
+        SNYK_EXIT_CODE=0
+        SOURCE_CODE_DIR=/var/workdir
+        SEVERITY_THRESHOLD="high"
+        if [ "${IMP_FINDINGS_ONLY}" == "false" ]; then
+          SEVERITY_THRESHOLD="low"
+        fi
+        set +e
+        # We do want to expand ARGS (it can be multiple CLI flags, not just one)
+        # shellcheck disable=SC2086
+        snyk code test $ARGS --severity-threshold="$SEVERITY_THRESHOLD" "$SOURCE_CODE_DIR" --max-depth=1 --sarif-file-output="${SOURCE_CODE_DIR}"/sast_snyk_check_out.json 1>&2 >>stdout.txt
+        SNYK_EXIT_CODE=$?
+        set -e
+        test_not_skipped=0
+        SKIP_MSG="We found 0 supported files"
+        grep -q "$SKIP_MSG" stdout.txt || test_not_skipped=$?
+
+        # In order to generate csdiff/v1, we need to add the whole path of the source code as Snyk only provides an URI to embed the context
+        csgrep --mode=json --embed-context=3 "${SOURCE_CODE_DIR}"/sast_snyk_check_out.json |
```

Comment:
@kdudka After talking with @zregvart we found that modifying the place of the Snyk results file was better later to generate the `oci-ta` task, as the `generate-ta-task.sh` script had some problems parsing the paths and wasn't working as expected.
What do you think? I tested it and it is working as expected: https://konflux.apps.stone-prod-p02.hjvn.p1.openshiftapps.com/application-pipeline/workspaces/jperezde/applications/test-coverity/pipelineruns/osh-cli-container-konflux-test-2-on-pull-request-4858d

Reply:
Unfortunately, this does not work as expected.  We need to change CWD for `csgrep --embed-context` to work.

I downloaded scan results from the above test and the source code context and, consequently, the `csdiff/v1` fingerprints were missing:
```
% oras pull quay.io/redhat-user-workloads/jperezde-tenant/test-coverity/osh-cli-container-konflux-test-2@sha256:c4035bff42739b9fefc53574a0d0ff3aa04a4758999d0a930a571fecfb49bc6d --registry-config authfile.json 
✓ Pulled      sast_snyk_check_out.sarif                                                                                                                           92.7/92.7 kB 100.00%  101ms
  └─ sha256:e6b94aa3738a92fd8e92532e1d445f103e463d898dcef5a81c7007eeb58abf56
✓ Pulled      application/vnd.oci.image.manifest.v1+json                                                                                                            741/741  B 100.00%   81µs
  └─ sha256:c4035bff42739b9fefc53574a0d0ff3aa04a4758999d0a930a571fecfb49bc6d
Pulled [registry] quay.io/redhat-user-workloads/jperezde-tenant/test-coverity/osh-cli-container-konflux-test-2@sha256:c4035bff42739b9fefc53574a0d0ff3aa04a4758999d0a930a571fecfb49bc6d
Digest: sha256:c4035bff42739b9fefc53574a0d0ff3aa04a4758999d0a930a571fecfb49bc6d

% grep csdiff/ sast_snyk_check_out.sarif
                        "csdiff/v0": "590b97142a3743e2f9da8306af9c76ea2dd5ff8d"
                        "csdiff/v0": "590b97142a3743e2f9da8306af9c76ea2dd5ff8d"
                        "csdiff/v0": "2b5f35e7e512ce4ab6cdb5a8fd3892af7f22fed2"
                        "csdiff/v0": "2b5f35e7e512ce4ab6cdb5a8fd3892af7f22fed2"
                        "csdiff/v0": "2b5f35e7e512ce4ab6cdb5a8fd3892af7f22fed2"
                        "csdiff/v0": "2b5f35e7e512ce4ab6cdb5a8fd3892af7f22fed2"
                        "csdiff/v0": "43d83de159411782c26f5ee322e843d474e0e662"
                        "csdiff/v0": "43d83de159411782c26f5ee322e843d474e0e662"
                        "csdiff/v0": "cd1b0cf57ce214aad5256237fd9608f395798d4d"
                        "csdiff/v0": "cd1b0cf57ce214aad5256237fd9608f395798d4d"
                        "csdiff/v0": "8ffe380c399394cd31318bae191b48edc7ab9a30"
                        "csdiff/v0": "e5902a94c4e13072c0b1dc1c125efdc4f86a1aaa"
                        "csdiff/v0": "8c4be3b0e43c0df08cafbe5e007f4537138d6424"
                        "csdiff/v0": "c4427807cc019b5e2266002d1f413d9c5485781c"
                        "csdiff/v0": "299e01d53746ef91c9db6d546629559ff257b186"
                        "csdiff/v0": "8f59b6bbbeb96b88e6ded4bfbe40d5f3f435fff9"
                        "csdiff/v0": "e949d747563df6f42e67a03e8dcb680fd3fd3ea2"
                        "csdiff/v0": "299360d3509b49ff70567e75f970d37cc79c4692"
                        "csdiff/v0": "759db9a2851f60a5e45eba2ce9ab5f51900f949d"
                        "csdiff/v0": "964f33c91777c5c9dfa1677a8ba10242527e2f3d"
                        "csdiff/v0": "66b1b6558c13df870dc1cdf962f42c68dcc59e30"
                        "csdiff/v0": "10464c70caaf9905d10f5018a5f45cfc243a5b62"
                        "csdiff/v0": "3656880aadb13d28ec5a5760d335e9cdd006a484"
                        "csdiff/v0": "3656880aadb13d28ec5a5760d335e9cdd006a484"
                        "csdiff/v0": "1dc7782c811b83befb57afa9c05b2a1fd21ba1f3"
                        "csdiff/v0": "1dc7782c811b83befb57afa9c05b2a1fd21ba1f3"
                        "csdiff/v0": "932cb88e28f436df7187e81f7d9baf72ca5527e6"
                        "csdiff/v0": "932cb88e28f436df7187e81f7d9baf72ca5527e6"
```

Reply:
Fixed: https://konflux.apps.stone-prod-p02.hjvn.p1.openshiftapps.com/application-pipeline/workspaces/jperezde/applications/test-coverity/pipelineruns/osh-cli-container-konflux-test-2-on-pull-request-cwjmv

```
% oras pull quay.io/redhat-user-workloads/jperezde-tenant/test-coverity/osh-cli-container-konflux-test-2@sha256:c16b783aebf787a36e6a17c104afd0d44dad3a9059d52a95291d85b6092021d6
✓ Pulled      sast_snyk_check_out.sarif                                                                                                                                                                               173/173 kB 100.00%   48ms
  └─ sha256:83a59f11360df0dde3f50a4da3d2eba446a2c4008c8558819449b90bdce8aa04
✓ Pulled      application/vnd.oci.image.manifest.v1+json                                                                                                                                                              742/742  B 100.00%  501µs
  └─ sha256:c16b783aebf787a36e6a17c104afd0d44dad3a9059d52a95291d85b6092021d6
Pulled [registry] quay.io/redhat-user-workloads/jperezde-tenant/test-coverity/osh-cli-container-konflux-test-2@sha256:c16b783aebf787a36e6a17c104afd0d44dad3a9059d52a95291d85b6092021d6
Digest: sha256:c16b783aebf787a36e6a17c104afd0d44dad3a9059d52a95291d85b6092021d6


% grep csdiff/ sast_snyk_check_out.sarif
                        "csdiff/v0": "2b5f35e7e512ce4ab6cdb5a8fd3892af7f22fed2",
                        "csdiff/v1": "dc7f797833032fdc064541034c2e958e8e95c656"
                        "csdiff/v0": "2b5f35e7e512ce4ab6cdb5a8fd3892af7f22fed2",
                        "csdiff/v1": "41fc798a50d6328b2ad099983989f8ad8ffc14c7"
                        "csdiff/v0": "2b5f35e7e512ce4ab6cdb5a8fd3892af7f22fed2",
                        "csdiff/v1": "70472ff8830d244f66f940bce9def891a5dcfef7"
                        "csdiff/v0": "2b5f35e7e512ce4ab6cdb5a8fd3892af7f22fed2",
                        "csdiff/v1": "68fb90019efbb3d31917bd313f89e017fbc736bc"
                        "csdiff/v0": "590b97142a3743e2f9da8306af9c76ea2dd5ff8d",
                        "csdiff/v1": "93a8a1a32ed6e15b2f99d02151fb17b722d1216e"
                        "csdiff/v0": "590b97142a3743e2f9da8306af9c76ea2dd5ff8d",
                        "csdiff/v1": "4b7c975c13f16a2551d66b5ed1db165bcbd69d7a"
                        "csdiff/v0": "43d83de159411782c26f5ee322e843d474e0e662",
                        "csdiff/v1": "8999b23d63627bb8723d69deff6ac19f2c5a61ac"
                        "csdiff/v0": "43d83de159411782c26f5ee322e843d474e0e662",
                        "csdiff/v1": "4c017eb2db2e341b1b6dba7befa5dd8fecf65529"
                        "csdiff/v0": "8c4be3b0e43c0df08cafbe5e007f4537138d6424",
                        "csdiff/v1": "850e4bd843625cb718518af0cace0266398683b6"
                        "csdiff/v0": "8ffe380c399394cd31318bae191b48edc7ab9a30",
                        "csdiff/v1": "5b61f9beb51cf9717f1a168958a6907417d91eb3"
                        "csdiff/v0": "e5902a94c4e13072c0b1dc1c125efdc4f86a1aaa",
                        "csdiff/v1": "d9ad35b868453a84984c5fcaa2f3d8f84a005dee"
                        "csdiff/v0": "3656880aadb13d28ec5a5760d335e9cdd006a484",
                        "csdiff/v1": "28f65eab462d4354f7fe4de51c391432d14906c4"
```